### DISCUSSION TITLE
Let's Encrypt Expiration email

### DISCUSSION THREAD

**Discussion by dinki** [dani-garcia/vaultwarden#2640] [Q&A] [answered]
Hey guys.  been using Vaultwarden for a short time now and really like it.  I'll preface this with I know just enough to make me dangerous.  I'm running under Docker using the default options and I'm using Cloudflare to forward to my machine via remote proxy using Caddy.  All of this works fine but suddenly I'm getting emails from Let's Encrypt saying my certificate is about to expire.  All of my other Docker containers are using the Cloudflare SSL certificate which doesn't have to be renewed.  

Can someone tell me if I need to do something special to use the Cloudflare certificate?  Thanks for your help and patience.

**Comment by Masgalor** [comment, marked as answer]
Hey @dinki .
Your question is not that easy to answer, because there are several possible scenarios.
I wil go with the most common one, so correct me if something sounds wrong to you.

You have one cert securing the connection between the user an cloudflare and one to secure the connection between cloudflare and your server. I guess you've used the docker-compose from the wiki which includes caddy with automatic certs from lets encrypt. So the cert we are talking about is the one between your server and cloudflare.

The good thing is, your cert will be renewed automatically so don't worry. The bad thing is, this doesn't answer your question.
If you want to use a cert from cloudfare for this connection, you need to disable automatic certs within caddy and import the certs from cloudflare. Unfortunately I use neither caddy nor cloudflare so I can't give you instructions for this. Follow the link below to the cloudflare documentation about their origin-ca, you might already know this since you have used it before, just in case.

https://developers.cloudflare.com/ssl/origin-configuration/origin-ca/

**Comment by dinki** [comment]
@Masgalor Thank you for taking the time to write this detailed answer.  My biggest concern is a gap in accessibility for my passwords but your statement of auto renew is a relief.  I certainly look into the Cloudflare information you provided and see if I want to implement that at some point.
